internal/manifest/exec.go

```diff
@@ -59,11 +59,15 @@ func executeAction(ctx context.Context, action actions.Action, execContext *exec
 		pid, err := execContext.resolveProjectID(ctx, action.ProjectKey)
 		if err != nil {
 			return nil, err
+		} else if !pid.IsValid() {
```

Comment:
maybe it's better to handle this case in execContext.resolve*?
e.g. return skderrors.NotFound instead of invalid project and invalid integration?

Reply:
execContext is using resolver and I recall that I handled many of those cases in resolver recently and for many objects resolver is already returning NotFound. But this shoudl be extended for integrations, since integrations handled differently in resolver - see `resolver.translateError`

Reply:
Agree with @ig-ra that only error checking. More than that `IsValid` only checks that it's not the zero value, it does not check the database that there is such object.

Reply:
there are some stuff that rely on it returning nil/invalid. we need to make a pass over all of these and settle on a convention.